src/serializers/infer.rs

```diff
@@ -405,23 +383,6 @@ pub(crate) fn infer_serialize_known<S: Serializer>(
         }};
     }
 
-    macro_rules! serialize_dict {
```

Comment:
same for serde/json, we could probably have got away with keeping the macro, but I thought it was clearer to have a separate function.